### PR TITLE
add chunk alloc system variable

### PR DESCRIPTION
**Pull request by keeplearning20221**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add system variable tidb_enable_reuse_chunk 、tidb_max_reuse_chunk、tidb_max_reuse_column、last_sql_use_alloc for chunk alloc pr
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -293,6 +293,12 @@ mysql> SELECT * FROM t1;
 - 默认值：`OFF`
 - 这个变量用来显示上一个 `execute` 语句所使用的执行计划是不是直接从 plan cache 中取出来的。
 
+### `last_sql_use_alloc` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION
+- 默认值：`OFF`
+- 这个变量用来显示上一个语句是否使用了 chunk alloc 。
```
Comment (tiancaiamao) [inline system-variables.md:300]: 只读还是用户能改的?

Comment (keeplearning20221) [reply system-variables.md:300]: 只读

File: system-variables.md [hunk 2]
```diff
@@ -1925,6 +1931,22 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 该变量默认值为 `CANCEL`，但在 TiDB v4.0.2 及之前的版本中，默认值为 `LOG`。
 - 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`oom-action`) 进行配置，升级到 v6.1.0 时会自动继承原有设置。
 
+### `tidb_max_reuse_chunk` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`128`
+- 可选值：`[0, 2147483647]`
+- 该变量控制 chunk alloc 最多缓存的 chunk 对象数,配置的过大可能导致OOM。
```
Comment (tiancaiamao) [inline system-variables.md:1940]: s/,/，

File: system-variables.md (hunk 1 above)
Comment (tiancaiamao) [reply system-variables.md:300]: 可以加上这个说明不?

File: system-variables.md (hunk 2 above)
Comment (keeplearning20221) [reply system-variables.md:1940]: 老师我看上下数据范围都是用 ,来表示的
image

File: system-variables.md [hunk 3]
```diff
@@ -2058,6 +2080,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 > * [Connector/J](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html) (`allowMultiQueries`)
 > * PHP [mysqli](https://dev.mysql.com/doc/apis-php/en/apis-php-mysqli.quickstart.multiple-statement.html) (`mysqli_multi_query`)
 
+### `tidb_enable_reuse_chunk` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 可选值：`OFF`，`ON`
+- 改变量用于控制 TiDB 是否启用 Chunk 对象缓存，如果为 ON 优先使用缓存中的对象不够才会从系统中申请，如果为OFF则直接从系统内存中申请。
```
Comment (tiancaiamao) [inline system-variables.md:2089]: OFF 中英混合时要排版要带上空格

File: system-variables.md (hunk 2 above)
Comment (tiancaiamao) [reply system-variables.md:1940]: 这里是一个英文的逗号，要用中文的逗号...你看其它地方都是

**Review by tiancaiamao — COMMENTED** [review]
Please fix the typo, rest LGTM

File: system-variables.md (hunk 3 above)
Comment (keeplearning20221) [reply system-variables.md]: done

File: system-variables.md (hunk 2 above)
Comment (keeplearning20221) [reply system-variables.md]: done

File: system-variables.md (hunk 1 above)
Comment (keeplearning20221) [reply system-variables.md]: done

**Review by tiancaiamao — APPROVED** [review]

File: system-variables.md [hunk 4]
```diff
@@ -293,6 +293,12 @@ mysql> SELECT * FROM t1;
 - 默认值：`OFF`
 - 这个变量用来显示上一个 `execute` 语句所使用的执行计划是不是直接从 plan cache 中取出来的。
 
+### `last_sql_use_alloc` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION
+- 默认值：`OFF`
+- 这个变量是一个只读变量，用来显示上一个语句是否使用了 Chunk allocation。
```
Comment (Oreoxmt) [inline system-variables.md:300]: 什么是 chunk allocation?

File: system-variables.md [hunk 5]
```diff
@@ -1925,6 +1931,22 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 该变量默认值为 `CANCEL`，但在 TiDB v4.0.2 及之前的版本中，默认值为 `LOG`。
 - 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`oom-action`) 进行配置，升级到 v6.1.0 时会自动继承原有设置。
 
+### `tidb_max_reuse_chunk` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`64`
+- 范围：`[0, 2147483647]`
+- 该变量控制 Chunk allocation 最多缓存的 Chunk 对象数。配置过大会增加 OOM 的风险。
```
Comment (Oreoxmt) [inline system-variables.md:1940]: "配置过大"说法很模糊，可以提供测试数据说明在设置大于某个值的时候会出现 OOM 吗

File: system-variables.md [hunk 6]
```diff
@@ -2058,6 +2080,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 > * [Connector/J](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html) (`allowMultiQueries`)
 > * PHP [mysqli](https://dev.mysql.com/doc/apis-php/en/apis-php-mysqli.quickstart.multiple-statement.html) (`mysqli_multi_query`)
 
+### `tidb_enable_reuse_chunk` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 可选值：`OFF`，`ON`
+- 该变量用于控制 TiDB 是否启用 Chunk 对象缓存。如果为 `ON` 则优先使用缓存中的对象，不够时才会从系统内存中申请。如果为 `OFF` 则直接从系统内存中申请。
```
Comment (Oreoxmt) [inline system-variables.md:2089]: “**不够**时才会从系统内存中申请”不够是指缓存为空还是缓存小于某个值呢

Comment (keeplearning20221) [reply system-variables.md:2089]: 缓存中找不到申请的对象时，会从内存中申请

Comment (Oreoxmt) [inline system-variables.md:2089]:
```suggestion
- 该变量用于控制 TiDB 是否启用 Chunk 对象缓存。如果为 `ON` 则优先使用缓存中的对象，缓存中找不到申请的对象时才会从系统内存中申请。如果为 `OFF` 则直接从系统内存中申请。
```

File: system-variables.md (hunk 4 above)
Comment (keeplearning20221) [reply system-variables.md:300]: 就是使用了chunk 复用中的内存对象

File: system-variables.md (hunk 5 above)
Comment (keeplearning20221) [reply system-variables.md:1940]: 估算了一下 1024 个columns 对象大概占用120M内存，这个参数是连接级别可能每个连接都受到影响了

File: system-variables.md (hunk 4 above)
Comment (Oreoxmt) [inline system-variables.md:300]:
```suggestion
- 这个变量是一个只读变量，用来显示上一个语句是否使用了缓存的 Chunk 对象 (Chunk allocation)。
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: system-variables.md [hunk 7]
```diff
@@ -1925,6 +1931,22 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 该变量默认值为 `CANCEL`，但在 TiDB v4.0.2 及之前的版本中，默认值为 `LOG`。
 - 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`oom-action`) 进行配置，升级到 v6.1.0 时会自动继承原有设置。
 
+### `tidb_max_reuse_chunk` <span class="version-mark">从 v6.4.0 版本开始引入</span>
```
Comment (shichun-0415) [inline system-variables.md:1934]: 排序需跟英文一致，`tidb_enable_reuse_chunk` 放前面，`tidb_max_reuse_chunk` 和 `tidb_max_reuse_column` 放后面。

File: system-variables.md [hunk 8]
```diff
@@ -2058,6 +2080,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 > * [Connector/J](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-configuration-properties.html) (`allowMultiQueries`)
 > * PHP [mysqli](https://dev.mysql.com/doc/apis-php/en/apis-php-mysqli.quickstart.multiple-statement.html) (`mysqli_multi_query`)
 
+### `tidb_enable_reuse_chunk` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`ON`
+- 可选值：`OFF`，`ON`
+- 该变量用于控制 TiDB 是否启用 Chunk 对象缓存。如果为 `ON` 则优先使用缓存中的对象，缓存中找不到申请的对象时才会从系统内存中申请。如果为 `OFF` 则直接从系统内存中申请。
```
Comment (shichun-0415) [inline system-variables.md:2089]:
```suggestion
- 该变量用于控制 TiDB 是否启用 Chunk 对象缓存。如果为 `ON`，则优先使用缓存中的 Chunk 对象，缓存中找不到申请的对象时才会从系统内存中申请。如果为 `OFF`，则直接从系统内存中申请 Chunk 对象。
```

**Comment by Oreoxmt** [comment]
DNM until https://github.com/pingcap/tidb/pull/38607 is merged

File: tidb-configuration-file.md [hunk 9]
```diff
@@ -15,6 +15,20 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：true
 + 如果需要创建大量的表（例如 10 万张以上），建议将此参数设置为 false。
 
+### `tidb_max_reuse_chunk` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
++ 用于控制每个 connection 最多缓存的 chunk 对象数。配置过大会增加 OOM 的风险。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:20]:
```suggestion
+ 用于控制每个连接最多缓存的 Chunk 对象数。配置过大会增加 OOM 的风险。
```

File: tidb-configuration-file.md [hunk 10]
```diff
@@ -15,6 +15,20 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 默认值：64
++ 最小值：0
++ 最大值：2147483647
+
+### `tidb_max_reuse_column` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
++ 用于控制每个 connection 最多缓存的 column 对象数。配置过大会增加 OOM 的风险。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:27]:
```suggestion
+ 用于控制每个连接最多缓存的 column 对象数。配置过大会增加 OOM 的风险。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by shichun-0415 — APPROVED** [review]

File: system-variables.md [hunk 11]
```diff
@@ -1912,6 +1945,14 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 范围：`[32, 2147483647]`
 - 这个变量用来设置执行过程中一个 chunk 最大的行数，设置过大可能引起缓存局部性的问题。
 
+### `tidb_max_delta_schema_count`
```
Comment (qiancai) [inline system-variables.md:1948]: @keeplearning20221  请问这个变量是从哪个 TiDB 版本开始引入的呢

**Pull request merged**